### ISSUE TITLE
Safety breach: large files would go through the network

### ISSUE THREAD

**Issue by brunobord** [vole/vole#55]
At the moment, nothing can be prevented against this "hack":
- I'm taking a very large file, encoding it into a very very large base64 string.
- building a vole-compliant JSON file that I'm putting in my local directory.

It would fill up people's vole directories, for nothing (it wouldn't be readable).
As a consequence, the web application would be broken too, because it would try to display a very very large string made of blurb.

What could be done to prevent this?

**Comment by daks** [comment]
That rises a good question: is actually only btsync managing the date sent to vole, or is there any control by the application?

**Comment by marktheunissen** [comment]
We can write a sanity check into Vole - "is the JSON file I'm about to open > 30k?" If it is, then don't open it. 

Can we stop people from adding gigabytes of garbage into their Vole folder? Not really. We'd need a way to tell the BitTorrent Sync app to only transfer files of a certain size. However, perhaps we want to allow video to be distributed this way...?

But anyway, if I'm following someone and they start pushing gigs of data to me... they will be swiftly unfollowed. Nobody who cares about their followers would do this. People who spam on twitter get unfollowed, for example.

**Comment by brunobord** [comment]
@marktheunissen 

> “if I'm following someone and they start pushing gigs of data to me... they will be swiftly unfollowed”

I agree.

**Comment by marktheunissen** [comment]
Vole can also monitor folder size and warn the user in the web interface when this happens. It could also automatically unfollow as soon as it reaches a limit and ask the user to review.

**Comment by daks** [comment]
Do Vole plan to keep using bittorent as the sync protocol, or is this just a way to get-started fast?

**Comment by marktheunissen** [comment]
We've reached out to the BitTorrent team and will be discussing all our options with them very soon. :) BTSync is a great technology and I'd love to work with them.

But, I'd also like Vole to work with a standardized, pluggable backend API - so that anyone can use any sync backend if they so choose - Dropbox, Amazon Sync, ownCloud, etc. The problem is that these don't give the awesome benefits of the torrent protocol.

**Comment by daks** [comment]
"But anyway, if I'm following someone and they start pushing gigs of data to me... they will be swiftly unfollowed. Nobody who cares about their followers would do this. People who spam on twitter get unfollowed, for example."

I'm okay for spam, but for security issues the application must do as much as posible to protect users from it.

**Comment by Fil** [comment]
I think this issue is certainly relevant for BTSync, but not for Vole.

**Comment by marktheunissen** [comment]
@daks Yes, I agree. For example, when we Unmarshal any JSON, we must be extra careful to prevent buffer overflows and strictly validate any input. Everything in the user's folder must be treated as potentially hostile.
